src/Sparrow/Utils/NativeMemory.cs

```diff
@@ -38,11 +38,20 @@ public class ThreadStats
             public ulong UnmanagedThreadId;
             public long Allocations;
             public long ReleasesFromOtherThreads;
-            private Thread _threadInstance;
+            private WeakReference<Thread> _threadInstance = new WeakReference<Thread>(null);
             public readonly int ManagedThreadId;
             private string _lastName = "Unknown";
-            public string Name => _threadInstance?.Name ?? _lastName;
-
+            public string Name
+            {
+                get
+                {
+                    if (_threadInstance.TryGetTarget(out var thread))
```

Comment:
I don't like this. You missed `_lastName`

Reply:
In addition to _lastName, is there anything else you want to be changed? Becuase this change _must_ happen, without it we have Thread objects leak.